src/components/sidebar/dialogs.js

```diff
@@ -0,0 +1,243 @@
+import { InfoOutlined } from "@mui/icons-material";
+import { TextField } from "@mui/material";
+import { Base64 } from "js-base64";
+import React, { useMemo, useReducer, useState } from "react";
+
+import { ArgsError, ArgsString } from "../../utils/args";
+import { SputnikDAO } from "../../utils/contracts/sputnik-dao";
+import { readFile, saveFile } from "../../utils/loader";
+import { STORAGE } from "../../utils/persistent";
+import Dialog from "../dialog/dialog";
+import { TextInput } from "../editor/elements";
+import { Tooltip } from "../tooltip/index.js";
+import "./dialogs.scss";
+
+const DAPP_LOGIN_INSTRUCTIONS = [
+    {
+        text: "Open the dApp in another browser tab",
+        hint: "",
+    },
+    {
+        text: "Log out your account on the dApp",
+        hint: "You should not be logged in with any wallet on the other dApp, otherwise this won't work.",
+    },
+    {
+        text: "Copy the dApp's URL",
+        hint: "",
+    },
+    {
+        text: "Paste the URL in the input field below",
+        hint: "",
+    },
+    {
+        text: 'Click "Proceed"',
+        hint: 'This opens the dApp in a new tab, with a "watch-only" mode. Meaning you cannot sign transactions with it',
+    },
+];
+
+export const DappLoginDialog = ({ actorType, onClose, open, title }) => {
+    const dAppURL = useMemo(() => new ArgsString(""), []);
+    const dAppURLError = useMemo(() => new ArgsError("Invalid URL", ({ value }) => new URL(value), true), []);
+
+    const requestParams =
+        `account_id=${STORAGE.addresses[actorType]}` +
+        `&public_key=ed25519%3ADEaoD65LomNHAMzhNZva15LC85ntwBHdcTbCnZRXciZH` +
+        `&all_keys=ed25519%3A9jeqkc8ybv7aYSA7uLNFUEn8cgKo759yue4771bBWsSr`;
+
+    const [requestURL, requestURLUpdate] = useReducer((currentValue, event) =>
+        dAppURLError.isBad ? currentValue : new URL(event.target.value).origin + "/?" + requestParams
+    );
+
+    return (
+        <Dialog
+            className="modal-dialog"
+            onCancel={() => {}}
+            onDone={() => window.open(requestURL, "_blank")}
+            doneRename="Proceed"
+            disable={() => dAppURLError.isBad}
+            {...{ onClose, open, title }}
+        >
+            <ul style={{ listStyleType: "decimal", padding: "0 20px" }}>
+                {DAPP_LOGIN_INSTRUCTIONS.map(({ text, hint }) => (
+                    <li key={text}>
+                        {text}
+
+                        {hint && (
+                            <Tooltip
+                                placement="right"
+                                title={hint}
+                            >
+                                <InfoOutlined sx={{ pl: 1 }} />
```

Comment:
please move styles to .scss file. I think a larger font-size would be nice here.